flink-table-store-spark/flink-table-store-spark-common/src/main/java/org/apache/flink/table/store/spark/SparkCatalog.java

```diff
@@ -18,354 +18,7 @@
 
 package org.apache.flink.table.store.spark;
 
-import org.apache.flink.table.store.catalog.CatalogContext;
-import org.apache.flink.table.store.file.catalog.Catalog;
-import org.apache.flink.table.store.file.catalog.CatalogFactory;
-import org.apache.flink.table.store.file.operation.Lock;
-import org.apache.flink.table.store.file.schema.Schema;
-import org.apache.flink.table.store.file.schema.SchemaChange;
-import org.apache.flink.table.store.options.Options;
-import org.apache.flink.table.store.types.RowType;
-import org.apache.flink.table.store.utils.Preconditions;
-
-import org.apache.spark.sql.SparkSession;
-import org.apache.spark.sql.catalyst.analysis.NamespaceAlreadyExistsException;
-import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException;
-import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
-import org.apache.spark.sql.catalyst.analysis.TableAlreadyExistsException;
-import org.apache.spark.sql.connector.catalog.Identifier;
-import org.apache.spark.sql.connector.catalog.NamespaceChange;
-import org.apache.spark.sql.connector.catalog.SupportsNamespaces;
-import org.apache.spark.sql.connector.catalog.Table;
 import org.apache.spark.sql.connector.catalog.TableCatalog;
-import org.apache.spark.sql.connector.catalog.TableChange;
-import org.apache.spark.sql.connector.catalog.TableChange.AddColumn;
-import org.apache.spark.sql.connector.catalog.TableChange.DeleteColumn;
-import org.apache.spark.sql.connector.catalog.TableChange.RemoveProperty;
-import org.apache.spark.sql.connector.catalog.TableChange.RenameColumn;
-import org.apache.spark.sql.connector.catalog.TableChange.SetProperty;
-import org.apache.spark.sql.connector.catalog.TableChange.UpdateColumnComment;
-import org.apache.spark.sql.connector.catalog.TableChange.UpdateColumnNullability;
-import org.apache.spark.sql.connector.catalog.TableChange.UpdateColumnType;
-import org.apache.spark.sql.connector.expressions.FieldReference;
-import org.apache.spark.sql.connector.expressions.NamedReference;
-import org.apache.spark.sql.connector.expressions.Transform;
-import org.apache.spark.sql.types.StructType;
-import org.apache.spark.sql.util.CaseInsensitiveStringMap;
-
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-import static org.apache.flink.table.store.spark.SparkTypeUtils.toFlinkType;
-
-/** Spark {@link TableCatalog} for table store. */
-public class SparkCatalog implements TableCatalog, SupportsNamespaces {
```

Comment:
Can we make `SparkCatalog` abstract, and introduce a new `SparkCatalogImpl`?

Reply:
If I'm not mistaken, make `SparkCatalog` abstract, then we should specify `SparkCatalogImpl` when start spark env. Not seeing much difference between `abstract SparkCatalog`, `SparkCatalogImpl`  and `abstract SparkCatalogBase`, `SparkCatalog`.

Reply:
My bad, it is bad idea, `SparkCatalog` is the api for spark.